doc/source/authentication.rst

```diff
@@ -83,16 +83,30 @@ tape archive, public cloud, or your own laptop. Start a Globus app
           callbackUrl: "https://<your_jupyterhub_host>/hub/oauth_callback"
           identityProvider: "youruniversity.edu"
 
+Keycloak (OpenID Connect)
+^^^^^^^^^^^^^^^^^^^^^^^^^
 
-To add a whitelist of usernames add to the config file under `auth`:
+OpenID Connect endpoint discovery is not supported by oauthentiator,
+but you can still configure JupyterHub to authenticate with it.
 
 .. code-block:: yaml
 
-     auth:
-         whitelist:
-             users:
-                - user1
-                - user2
+      hub:
+        extraEnv:
+          OAUTH2_AUTHORIZE_URL: https://${host}/auth/realms/${realm}/protocol/openid-connect/auth
+          OAUTH2_TOKEN_URL: https://${host}/auth/realms/${realm}/protocol/openid-connect/token
+      auth:
+        type: custom
+        custom:
+          className: oauthenticator.generic.GenericOAuthenticator
+          config:
+            client_id: "y0urc1logonc1ient1d"
```

Comment:
we usually use variable-naming like this `<YOUR-VARIABLE-HERE>`...could you do the same for the client_id and secret fields?

Reply:
sure, but do you mean host and realm? because i copied those stub client_id/secret from other sections.

Reply:
nah I just mean `client_id: "<YOUR-CLIENT-ID>"` and the same for the secret just beneath

Reply:
gah, actually, nevermind. I just realized all of our other examples use this too...just leave it the way it is and we should revise that in a separate PR for the whole page